### ISSUE TITLE
feature request: auto save files

### ISSUE THREAD

**Issue by grsjst** [SWI-Prolog/swish#37]
Hi Jan,

Swish is awesome! I have been using it as my editor and that worked great until a hiccup occurred (browser crash) and I found myself losing some work due to not having saved my latest edits. Of course I am the one to blame for this, but I realised auto saving is a welcome feature :)

thanks,
 joost

**Comment by JanWielemaker** [comment]
Hi Joost,

Glad you like it!

Makes sense.   I once had the idea to save the current state to the browser's local store when leaving
the page, so you can come back at the same state.  That could include autosave.

**Comment by grsjst** [comment]
Yes that could work. I am running swish on my local machine (there are no
other users than myself), so using the existing gitty infrastructure would
be ok in my case with the added advantage of having a history (if you are
hosting for multiple users this might be problematic though).
